### PR TITLE
Move releasing-airflow docs to dev/README.md

### LINKED ISSUE #10183: Move "Releasing Airflow" docs to the "dev"
**Description**

The documentation for [Releasing Airflow](https://cwiki.apache.org/confluence/pages/viewpage.action?pageId=69410380) should be reviewed and incorporated in [dev README.md](https://github.com/apache/airflow/blob/master/dev/README.md)

**Use case / motivation**

To remove documentation duplication and clean it up.

**Related Issues**

#10178

### PR DESCRIPTION
**Pull request by sergioesca4**
Moving releasing airflow docs to dev/README.md to remove documentation duplication and clean it up.
---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).

### CODE REVIEW

File: dev/README.md [hunk 1]
```diff
@@ -243,3 +251,331 @@ Here is a typical scenario:
...
+It is important to stress that this snapshot is not intended for users.
+
+# Make sure your public key is on id.apache.org and in KEYS
+
+You will need to sign the release artifacts with your pgp key. After you have created a key, make sure you
+
+- Add your GPG pub key to https://dist.apache.org/repos/dist/release/airflow/KEYS , follow the instructions at the top of that file. Upload your GPG public key to https://pgp.mit.edu 
+- Add your key fingerprint to https://id.apache.org/ (login with your apache credentials, paste your fingerprint into the pgp fingerprint field and hit save).
+
+```
```
Comment (potiuk) [inline dev/README.md]:
```suggestion
```shell script
```

Comment (potiuk) [reply dev/README.md]: And all the places below - this way we get proper syntax coloring

**Comment by potiuk** [comment]
looks good - just the syntax check :)

**Comment by kaxil** [comment]
Static checks are failing: https://github.com/apache/airflow/pull/11245/checks?check_run_id=1200805794

**Comment by potiuk** [comment]
Hey @sergioesca4 - I would like to work on adding some information about backport package releases. I am happy to rebase this one and fix the static check!

**Comment by potiuk** [comment]
I'v pushed the rebased/squashed changes and merge when it passes @sergioesca4 :)

**Review by potiuk — APPROVED** [review]

**Comment by potiuk** [comment]
Merged it now! Thanks @sergioesca4  -> I am refactoring it now to merge with Backport Packages. It's been realy helpful for that.!

**Pull request merged**